src/party.rs

```diff
@@ -24,67 +62,876 @@ pub struct BallotConfig {
 pub struct Party<V: Value, VS: ValueSelector<V>> {
     /// This party's identifier.
     pub id: u64,
-    /// Other ballot parties' ids.
-    pub party_ids: Vec<u64>,
 
     /// Communication queues.
-    in_receiver: Receiver<MessageWire>,
-    out_sender: Sender<MessageWire>,
+    msg_in_receiver: Receiver<MessageWire>,
+    msg_out_sender: Sender<MessageWire>,
 
-    /// Query to submit result.
-    value_sender: Sender<Result<V, BallotError>>,
+    /// Query to receive and send events that run ballot protocol
+    event_receiver: Receiver<PartyEvent>,
+    event_sender: Sender<PartyEvent>,
 
-    /// Ballot config (e.g. ballot time bounds).
-    cfg: BallotConfig,
+    /// BPCon config (e.g. ballot time bounds, parties weights, etc.).
+    cfg: BPConConfig,
 
     /// Main functional for value selection.
     value_selector: VS,
 
-    // TODO: define other state fields if needed.
+    /// Status of the ballot execution
+    status: PartyStatus,
+
+    /// Current ballot number
+    ballot: u64,
+
+    /// Last ballot where party submitted 2b message
+    last_ballot_voted: Option<u64>,
+
+    /// Last value for which party submitted 2b message
+    last_value_voted: Option<Vec<u8>>,
+
+    /// Local round fields
+
+    /// 1b round state
+    ///
+    parties_voted_before: HashMap<u64, Option<Vec<u8>>>, // id <-> value
+    messages_1b_weight: u128,
+
+    /// 2a round state
+    ///
+    value_2a: Option<V>,
+
+    /// 2av round state
+    ///
+    messages_2av_senders: HashSet<u64>,
+    messages_2av_weight: u128,
+
+    /// 2b round state
+    ///
+    messages_2b_senders: HashSet<u64>,
+    messages_2b_weight: u128,
```

Comment:
I might create a separate struct for these senders/weight pairs as the `Party` struct is getting quite large as it is -- that would let us reduce the struct fields by two at least.